### ISSUE TITLE
Dictionary-to-cloze-delection flashcard

### ISSUE THREAD

**Issue by lakesare** [lakesare/memcode#147]
When we create cloze deletion flashcards for some word we don't know, e.g. "**grit**", we frequently go through the same algorithm manually:  
1. Go to lexico.com: https://www.lexico.com/definition/grit
2. Copy a good example sentence
3. Copy the definition
4. Finally create the flashcard.

Is it possible to automate these tasks via the dictionary api?

___

<img width="970" alt="image" src="https://user-images.githubusercontent.com/7578559/154254762-9aad17bb-1a81-4929-99a3-ba31f22ce91c.png">

<img width="300" alt="image" src="https://user-images.githubusercontent.com/7578559/154254818-481f5987-dde6-4793-aa89-9bd44d8940b2.png">

**Comment by iopq** [comment]
I'm using Tatoeba for the same purpose, since it has Korean sentences

**Comment by lakesare** [comment]
@iopq, very interesting, thanks!

**Comment by iopq** [comment]
I tried an app called Clozemaster and it works like you suggested, it's pretty good, but freemium. For Korean it sentence mines Tatoeba.

The best feature is actually cloze cards with TTS, where you have to listen to the card's audio and input it. Really works your listening comprehension in context.

So after using it, I think this is a great idea. Even better if you can just import a bunch of cards without even choosing the word, like by certain parameters like difficulty. Then you can just make a deck, import a thousand sentences at your level and get cracking right away

**Comment by lakesare** [comment]
Yeah text-to-speech is very good for new languages; and already-existing flashcards are also pretty important for new languages, memcode lacks in that (I had to learn some german over the summer and it was evident memcode wasn't up to speed here; only good when you're already rather advanced).  

The first problem (getting to crunch words right away) is solved simply by creating courses. For example, we could have an admin-approved 500-most-used-words courses per each language.

The second problem (TTS) is something I could implement, but would require some time/funding.

**Comment by iopq** [comment]
The best part about Tatoeba is not the words, it's the sentences. You need to be a native to decide which sentences are culturally significant.

Example: Good morning is 좋은 아침이에요, but it is not used as a greeting in Korean. In the morning you still wish someone tranquility, just like any other time.

The sentence actually means the morning is good, so from the grammar standpoint it's not wrong.

Tatoeba also has recordings: 

https://tatoeba.org/en/sentences/show/2704960

None for Korean, but that's just an issue of nobody making them.

But let's say you want to make and maintain courses for languages, what's the UI for making those first 500 cards? It's 500 for each language translated to another language, so 500 per mapping

**Comment by lakesare** [comment]
Not words of course (imagine that!), I meant sentences with words cloze deleted/question-answered.

<img width="1054" alt="image" src="https://user-images.githubusercontent.com/7578559/213902036-32c009e2-27e7-4dd2-8afd-3fbd315c77bb.png">
<img width="1071" alt="image" src="https://user-images.githubusercontent.com/7578559/213902065-6931b980-9cd5-4f94-8906-8303943e7750.png">

I think there should exist some databases with such sentences, please @ me if you know of them.  
Having such courses sounds like a very good idea now that I'm thinking it over.

**Comment by iopq** [comment]
https://glitch.com/edit/#!/browser-voices?path=index.html%3A1%3A0

Before I forget, I got this to work on my phone. Requires enabling text to speech, but the Google one is pretty good

**Comment by lakesare** [comment]
Nice! I worried non-english accents might be bad with a free browser api, did you check it for Korean?

**Comment by iopq** [comment]
> Nice! I worried non-english accents might be bad with a free browser api, did you check it for Korean?

Korean sounds great, so does Chinese

There's just a weird bug where if you have language detection off it will read it in English, even if written in the Korean hangeul

**Comment by iopq** [comment]
![image](https://user-images.githubusercontent.com/1817528/215237526-85be89fa-ea43-4f02-8fdf-1410d20805b8.png)

this is a really cool feature for this type of card as well, since it looked up by selection (anything I select is automatically looked up), not by word - I just selected the answer and got the Chinese characters straight from wiktionary so I can relate the 차 to meaning of the character I already know, 次 which means "x times" while the whole word means more like "step by step"

the clicking part is not useful because wiktionary doesn't have the combinations with post-positions and full search just finds random articles with example sentences

although this is maybe a complete separate feature

**Comment by lakesare** [comment]
I have an extension in my browser for this (https://addons.mozilla.org/en-US/firefox/addon/dictionary-anyvhere/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search):

<img width="365" alt="image" src="https://user-images.githubusercontent.com/7578559/215237940-8539a4e1-56cf-4518-9c2a-478cd1976265.png">

Maybe this functionality is better as an extension.

**Comment by lakesare** [comment]
### Price assignment (re: [memcode#requesting-a-feature](https://github.com/lakesare/memcode#requesting-a-feature))

**Automating the creation of cloze-deletion cards via the dictionary**: **$150**
